docs/auth/add-auth-provider.md

```diff
@@ -0,0 +1,193 @@
+  const oauthProvider = new OAuthProvider(provider, config.provider, true);
+  return oauthProvider;
+}
+```
+
+Not extending with OAuth, the main difference here is that the create method is returning a instance of the class without adding the OAuth authorization framework to it.
+
+```ts
+export function createProviderAProvider(config: AuthProviderConfig) {
+  return new ProviderAAuthProvider(config);
+}
+```
+
+#### Verify Callback
+
+> Strategies require what is known as a verify callback. The purpose of a verify callback is to find the user that possesses a set of credentials.
+> When Passport authenticates a request, it parses the credentials contained in the request. It then invokes the verify callback with those credentials as arguments [...]. If the credentials are valid, the verify callback invokes done to supply Passport with the user that authenticated.
+>
+> If the credentials are not valid (for example, if the password is incorrect), done should be invoked with false instead of a user to indicate an authentication failure.
+>
+> http://www.passportjs.org/docs/configure/
+
+**`plugins/auth-backend/src/providers/providerA/index.ts`** is simply re-exporting the create method to be used for hooking the provider up to the backend.
+
+```ts
+export { createProviderAProvider } from './provider';
+```
+
+### Hook it up to the backend
+
+**`plugins/auth-backend/src/providers/config.ts`** The provider needs to be configured properly so you need to add it to the list of configured providers, all of which implement [AuthProviderConfig](#AuthProviderConfig):
+
+```ts
+export const providers = [
+  {
+    provider: 'providerA',  # used as an identifier
+    options: { ... },       # consult the provider documentation for which options you should provide
+    disableRefresh: true    # if the provider lacks refresh tokens
+  },
+```
+
+**`plugins/auth-backend/src/providers/factories.ts`** When the `auth-backend` starts it sets up routing for all the available providers by calling `createAuthProviderRouter` on each provider. You need to import the create method from the provider and add it to the factory:
+
+```ts
+import { createProviderAProvider } from './providerA';
+const factories: { [providerId: string]: AuthProviderFactory } = {
+  providerA: createProviderAProvider,
+};
+```
+
+By doing this `auth-backend` automatically adds these endpoints:
+
+```ts
+router.get('/auth/providerA/start');
+router.get('/auth/providerA/handler/frame');
+router.post('/auth/providerA/handler/frame');
+router.post('/auth/providerA/logout');
+router.get('/auth/providerA/refresh'); // if supported
+```
+
+As you can see each endpoint is prefixed with both `/auth` and its provider name.
+
+### Test the new provider
+
+You can `curl -i localhost:7000/auth/providerA/start` and which should provide a `302` redirect with a `Location` header. Paste the url from that header into a web browser and you should be able to trigger the authorization flow.
+
+### To summarize
```

Comment:
would it better if this was at the top and we link to the different sections?

Reply:
👍